Sources/HummingbirdXCT/ApplicationBuilder+XCT.swift

```diff
@@ -46,43 +46,22 @@ public enum XCTTestingSetup {
 ///     XCTAssertEqual(String(buffer: body, "hello")
 /// }
 /// ```
-extension HBApplication {
+extension HBApplicationBuilder {
     // MARK: Initialization
 
     /// Creates a version of `HBApplication` that can be used for testing code
     ///
     /// - Parameters:
     ///   - testing: indicates which type of testing framework we want
     ///   - configuration: configuration of application
-    public convenience init(testing: XCTTestingSetup, configuration: HBApplication.Configuration = .init(), timeout: TimeAmount = .seconds(15)) {
-        let xct: any HBXCT
-        let configuration = configuration.with(address: .hostname("localhost", port: 0))
+    __consuming public func buildAndTest(_ testing: XCTTestingSetup, _ test: @escaping @Sendable (any HBXCTClientProtocol) async throws -> Void) async throws {
```

Comment:
Marked to note the same question about `__consuming`

Reply:
see previous comment